Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,431 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+
+variable (F : WalkingReflexivePair ⥤ C)
+
+/-- Forgetting the reflexion yields an equivalence between cocones over a bundled reflexive pair and
+coforks on the underlying parallel pair. -/
+def inclusionWalkingReflexivePairEquivCocone :
+    Cocone F ≌ Cofork (F.map left) (F.map right) :=
+  (Functor.Final.coconesEquiv _ F).symm.trans
+    (Cocones.precomposeEquivalence
+      (diagramIsoParallelPair (WalkingParallelPair.inclusionWalkingReflexivePair ⋙ F))).symm
+
+@[inherit_doc inclusionWalkingReflexivePairEquivCocone]
+def inclusionWalkingReflexivePairEquiv_obj (G : ReflexiveCofork F) :
+    (inclusionWalkingReflexivePairEquivCocone F).functor.obj G ≅ G.toCofork :=
+  Cofork.ext (Iso.refl _)
+    (by simp [inclusionWalkingReflexivePairEquivCocone, Cofork.π])
+
+instance reflexivePair_hasColimit_of_hasCoequalizer
+    [h : HasCoequalizer (F.map left) (F.map right)] : HasColimit F := by
+  have := Limits.hasColimitOfIso (show _ ≅ parallelPair (F.map left) (F.map right) from
+    diagramIsoParallelPair (WalkingParallelPair.inclusionWalkingReflexivePair ⋙ F))
+  apply Functor.Final.hasColimit_of_comp WalkingParallelPair.inclusionWalkingReflexivePair
+
+lemma hasReflexiveCoequalizer_iff_hasCoequalizer :
+    HasColimit F ↔ HasCoequalizer (F.map left) (F.map right) := by
+  simpa only [hasColimit_iff_hasInitial_cocone]
+    using Equivalence.hasInitial_iff (inclusionWalkingReflexivePairEquivCocone F)
+
+/-- A reflexive cofork is a colimit cocone if and only if the underlying cofork is. -/
+def ReflexiveCofork.isColimitEquiv (G : ReflexiveCofork F) :
+    IsColimit (G.toCofork) ≃ IsColimit G :=
+  IsColimit.equivIsoColimit (inclusionWalkingReflexivePairEquiv_obj F G).symm|>.trans <|
+    (IsColimit.precomposeHomEquiv (diagramIsoParallelPair _).symm (G.whisker _)).trans <|
+      Functor.Final.isColimitWhiskerEquiv _ _
+
+/-- The colimit of a functor out of the walking reflexive pair is the same as the colimit of the
+underlying parallel pair. -/
+def inclusionWalkingReflexivePairColimitEquiv
+    [HasCoequalizer (F.map left) (F.map right)] :
+    colimit F ≅ coequalizer (F.map left) (F.map right) :=
+  ((ReflexiveCofork.isColimitEquiv _ _).symm (colimit.isColimit F)).coconePointUniqueUpToIso
+    (colimit.isColimit _)variable {A B : C} {f g : A ⟶ B} [IsReflexivePair f g]
+
+instance ofIsReflexivePairHasColimit_of_hasCoequalizer [HasCoequalizer f g] :
+    HasColimit (ofIsReflexivePair f g) := by
+  have := Limits.hasColimitOfIso <| inclusionWalkingReflexivePairOfIsReflexivePairIso f g
+  apply Functor.Final.hasColimit_of_comp WalkingParallelPair.inclusionWalkingReflexivePair
+
+/-- The coequalizer of a reflexive pair can be promoted to the colimit of a diagram out of the
+walking reflexive pair -/
+def ofIsReflexivePairColimitEquiv [HasCoequalizer f g] :
+    colimit (ofIsReflexivePair f g) ≅ coequalizer f g :=
+  (Functor.Final.colimitIso _ _).symm.trans <|
+    HasColimit.isoOfNatIso <| inclusionWalkingReflexivePairOfIsReflexivePairIso _ _
```

Comment:
The name should be changed. For the "proof", it would be better to apply `reflexiveCoequalizerIsoCoequalizer` to `ofIsReflexivePair f g` instead of proving it again.